report_templates/ansible_inventory.erb

```diff
@@ -107,7 +115,9 @@ require:
   version: '>= 3.9.0'
 -%>
 <%- content_attrs = input('Content Attributes') == 'yes' %>
-<%- load_hosts(search: input('Hosts'), includes: [:organization, :location, :hostgroup, :host_collections, { interfaces: :subnet }, :content_facet]).each_record do |host| -%>
+<%- includes = [:organization, :location, :hostgroup, :host_collections, { interfaces: :subnet }, :content_facet] %>
+<%- includes = [:organization, :location, :hostgroup, :host_collections, { interfaces: [:subnet, :domain] }, :content_facet, :operatingsystem] if input('Host Parameters') == 'yes' %>
```

Comment:
I'd prefer to either don't repeat the list

```ruby
interface_includes = [ :subnet ]
interface_includes.push(:domain) if input('Host Parameters') == 'yes'
includes = includes = [:organization, :location, :hostgroup, :host_collections, { interfaces: interface_includes }, :content_facet]
includes.push(:operatingsystem) if input('Host Parameters') == 'yes'
```

or don't define the variable twice
```ruby
if input('Host Parameters') == 'yes'
  includes = [:organization, :location, :hostgroup, :host_collections, { interfaces: [:subnet, :domain] }, :content_facet, :operatingsystem]
else
  includes = [:organization, :location, :hostgroup, :host_collections, { interfaces: :subnet }, :content_facet]
end  
```

the later may be more readable, but the first one more extendable if we want to have more conditions for other inputs in future

Reply:
I've chosen the first way, thanks :)